src/core_plugins/kibana/public/discover/controllers/discover.js

```diff
@@ -182,6 +182,17 @@ function discoverController(
     .highlightAll(true)
     .version(true);
 
+  // searchSource which applies time range
+  const timeRangeSearchSource = savedSearch.searchSource.clone();
```

Comment:
Do we actually want to `clone` here, and not just creating a new instance? I've got the feeling except the filter (and maybe the index) we don't want to have anything from that `savedSearch.searchSource`? Wouldn't it be easier to create a new one, and copy this two things over rather than cloning and resetting some things (like `query`) below, but might forget to reset some important things?

Reply:
what are the things you wouldn't want ? when i was thinking about it it was either:
- create a new one and make sure the parameters are set correctly (index ... ? not sure if anything else is needed ?)
- clone, and make sure the parameters are set correctly <-- sounded safer, as i don't need to worry about all parameters + i don't need to create a new SearchSource which means i don't need access to Private